File: packages/go/compliance.go
package opensms

import (
	"context"
	"net/http"
)

// Compliance reads country and content rules, reached as client.Compliance.
type Compliance struct {
	http *transport
}

// ListCountries returns the rules for every country
// (GET /v1/compliance/countries).
func (r *Compliance) ListCountries(ctx context.Context) ([]CountryRules, error) {
	return callSlice[CountryRules](ctx, r.http, request{method: http.MethodGet, path: "/v1/compliance/countries"})
}

// GetCountry returns one country's rules (GET /v1/compliance/countries/{iso2}).
func (r *Compliance) GetCountry(ctx context.Context, iso2 string) (*CountryRules, error) {
	if err := requireID("iso2", iso2); err != nil {
		return nil, err
	}
	return call[CountryRules](ctx, r.http, request{method: http.MethodGet, path: "/v1/compliance/countries/" + escape(iso2)})
}

// ListContentRules returns the platform content rules (GET /v1/content-rules).
func (r *Compliance) ListContentRules(ctx context.Context) ([]ContentRule, error) {
	return callSlice[ContentRule](ctx, r.http, request{method: http.MethodGet, path: "/v1/content-rules"})
}

File: packages/java/src/main/java/io/opensms/Compliance.java
package io.opensms;

import io.opensms.models.ContentRule;
import io.opensms.models.CountryRules;

import java.util.List;

/** Country compliance rules and content rules. Accessed as {@code client.compliance()}. */
public final class Compliance {

    private final ApiTransport t;

    Compliance(ApiTransport t) {
        this.t = t;
    }

    /** @return rules for every country. */
    public List<CountryRules> listCountries() {
        return t.get("/v1/compliance/countries", null, ApiTransport.listOf(CountryRules.class));
    }

    /**
     * @param iso2 ISO2 country.
     * @return that country's rules.
     */
    public CountryRules getCountry(String iso2) {
        return t.get("/v1/compliance/countries/" + Wire.id(iso2, "iso2"), null, ApiTransport.type(CountryRules.class));
    }

    /** @return every content rule. */
    public List<ContentRule> listContentRules() {
        return t.get("/v1/content-rules", null, ApiTransport.listOf(ContentRule.class));
    }
}
